### DISCUSSION TITLE
Compute Shaders (hlsl-files)

### DISCUSSION THREAD

**Discussion by llfab** [StereoKit/StereoKit#318] [Q&A]
We may plan to use compute shaders for April-Tag (https://april.eecs.umich.edu/software/apriltag) tracking with the Hololens front camera. Parts of the tracking algo are run on GPU using DirectX 12 compute shaders. Can I get the DX context via SK to add and run these compute shaders?

**Comment by maluoi** [comment]
SK doesn't currently surface the DirectX context, but that has been something I've considered. StereoKit uses DX11 though, so I don't know exactly how helpful that'd be for you? Currently, there is a `Backend` class with [OpenXR](https://stereokit.net/Pages/Reference/OpenXR.html), would something similar for DirectX work for you?

I have also been working on direct Compute shader support for SK, but that probably won't make it into the upcoming update.

**Reply by llfab** [reply]
Ok. Thanks. Also not sure whether it really is SK's task to provide means for GPU computing. Generally, I believe providing the pure backend "handles" is not such a bad idea. Yes, SK provides a lot of things very nicely in an abstract and convenient fashion. But every API will never provide the full possibilities of the abstracted backend. In this specific case thought, I am not so sure. As far as I understand, developers would prefer DX12 for compute shaders, hence, they would need to open an extra context anyways. 

So, in general allowing access to the DX11 context is a good idea if we find actual tasks that are solved with this.

In our specific case, we may potentially go for our own DX12 context.
